source/NPC.h

```diff
@@ -86,6 +93,13 @@ class NPC {
 	Conversation conversation;
 	const Conversation *stockConversation = nullptr;
 	
+	// Conditions that must be met in order for this NPC to be placed or despawned:
+	bool checkedSpawnConditions = false;
```

Comment:
Why not initialize both passed conditions with false and skip the checked variable? Then you can set passedSpawnConditions to true if the conditions-set is empty or if the test passes.